text/0000-build-std.md

```diff
@@ -0,0 +1,3601 @@
+depends on a crate `foo`, rustc will look in `-L` paths and the sysroot for it.
+
+All Cargo dependencies are provided to the compiler using the
+`--extern noprelude:` flag, including explicit and implicit standard library
+dependencies.
+
+*See the following sections for rationale/alternatives:*
+- [*Why prevent rustc from loading root dependencies from the sysroot?*][why-prevent-rustc-from-loading-root-dependencies-from-the-sysroot]
+- [*Why use `noprelude` with `--extern`?*][why-use-noprelude-with-extern]
+
+### Vendored `rust-src`
+[vendored-rust-src]: #vendored-rust-src
+
+When it is necessary to build the standard library, Cargo will look for sources
+in a fixed location in the sysroot: `lib/rustlib/src`. rustup's `rust-src`
+component downloads standard library sources to this location. If the sources
+are not found, Cargo will emit an error and recommend the user download
+`rust-src` if using rustup.
+
+`rust-src` will contain the sources for the standard library crates as well as
+its vendored dependencies. Sources of standard library dependencies will never
+be fetched from crates.io.
+
+Cargo will not perform any checks to ensure that the sources in `rust-src` have
+been modified. It will be documented that modifying these sources is not
+supported.
+
+*See the following sections for rationale/alternatives:*
+- [*Why not check if `rust-src` has been modified?*][why-not-check-if-rust-src-has-been-modified]
+- [*Why vendor standard library dependencies?*][why-vendor-the-standard-librarys-dependencies]
+- [*Why not allow the source path for the standard library be customised?*][why-not-allow-the-source-path-for-the-standard-library-be-customised]
+
+### Building the standard library on a stable toolchain
+[building-the-standard-library-on-a-stable-toolchain]: #building-the-standard-library-on-a-stable-toolchain
+
+rustc will automatically assume `RUSTC_BOOTSTRAP` when the source path of the
+crate being compiled is within the same sysroot as the rustc binary being
+invoked. Cargo will not need to use `RUSTC_BOOTSTRAP` when compiling the
+standard library with a stable toolchain.
+
+*See the following sections for rationale/alternatives:*
+- [*Why allow building from the sysroot with implied `RUSTC_BOOTSTRAP`?*][why-allow-building-from-the-sysroot-with-implied-rustc-bootstrap]
+
+### Panic strategies
+[panic-strategies]: #panic-strategies-1
+
+Panic strategies are unlike other profile settings insofar as they influence
+which crates and flags are passed to the standard library. For example. if
+`panic = "unwind"` were set in the Cargo profile then the `panic_unwind` feature
+would need to be provided to `std` and `-Cpanic=unwind` passed to suggest that
+the compiler use that panic runtime.
+
+If the current crate has no dependency on `std` (i.e. have added an `alloc` or
+`core` dependency explicitly to opt-out of the implicit `std` dependency), then
+Cargo will not build either of the `panic_unwind` or `panic_abort` crates or
```

Comment:
I think it's a valid use case to use these panic implementations in a no_std crate

Reply:
My mistake, all the examples I found were building std (because indeed std chooses the panic runtime to use)